Documentation/ceph-csi-troubleshooting.md

```diff
@@ -495,3 +495,85 @@ If the Ceph version is Octopus(v15.2.x) or older, run:
+E.g.:
+```text
+MountVolume.MountDevice failed for volume "pvc-272a60b6-a51a-455a-b269-c794e9629b83" : rpc error: code = Internal desc = rbd image replicapool/csi-vol-e23bc868-8788-11eb-ab3c-1a1f46590305 is still being used
+```
+And you are using `ReadWriteOnce` PV/PVCs. E.g.:
+```yaml
+spec:
+  accessModes:
+  - ReadWriteOnce
+```
+It likely indicates that the object is still mounted as an `rbd` device on a
+node other than where the pod is being scheduled.  One possible cause for this
+scenario is that the pod was successfully provisioned with a mounted volume and
+then drained from the node is was originally schedule upon.
+
+### Solution
+
+The `rbd` device needs to be unmounted **from the node upon which it is mounted**.
+
+First, identify the `csi-rbdplugin` pod which is mounting the object.  As there
+should be only one of these pods scheduled per node, we do not need to
+determine the actual node name.
+
+The method below for achieving this was provided in [this issue](https://github.com/rook/rook/issues/4772#issuecomment-601064683)
+
+```bash
+for pod in `kubectl -n rook-ceph get pods|grep rbdplugin|grep -v provisioner|awk '{print $1}'`; do
+  echo $pod
+  kubectl exec -it -n rook-ceph $pod -c csi-rbdplugin -- rbd device list
+done
+```
+
+The output should be similar to the following on an example three node cluster
+with only one rbd device.
+
+```bash
+$ for pod in `kubectl -n rook-ceph get pods|grep dplugin|grep -v provisioner|awk '{print $1}'`; do   echo $pod;   kubectl exec -it -n rook-ceph $pod -c csi-rbdplugin -- rbd device list;  done
+csi-rbdplugin-78wss
+csi-rbdplugin-rspgd
+csi-rbdplugin-z7hx6
+id  pool         namespace  image                                         snap  device
+0   replicapool             csi-vol-8a723bf8-e590-11eb-bb07-6adacddf2e23  -     /dev/rbd0
+```
+
+If we were looking for the `csi-vol-8a723bf8-e590-11eb-bb07-6adacddf2e23`
+object, we can see that its mount is visible in the `csi-rbdplugin-z7hx6`.
+
+Then enter into the appropriate `csi-rbdplugin` pod and umount or "unmap" the
+`rbd` device using the `rbd` utility. E.g.:
+
+```bash
+$ kubectl -n rook-ceph exec -ti csi-rbdplugin-z7hx6 -c csi-rbdplugin -- sh
+sh-4.4#
+sh-4.4# rbd device list
+id  pool         namespace  image                                         snap  device
+0   replicapool             csi-vol-8a723bf8-e590-11eb-bb07-6adacddf2e23  -     /dev/rbd0
+sh-4.4# umount /dev/rbd0
+sh-4.4# rbd unmap /dev/rbd0
+sh-4.4# rbd device list
+sh-4.4#
+```
+
+If `rbd umap` returns the following error:
+
+```bash
+sh-4.4# rbd unmap /dev/rbd0
+rbd: sysfs write failed
+rbd: unmap failed: (16) Device or resource busy
+```
+
+It means that `rbd` device is likely still in use by an active pod.
+
+**Note that even if the "toolbox pod" was scheduled on the node which has the
+active `rbd` mount, it does not commonly mount `/dev` from the host and will
+not be helpful in locating or resolving the undesired active mount.**
```

Comment:
this is not required if we identify all the mounts and do `umounts` properly on the node and at last unmap the device on the node?